### PR TITLE
Cropping faces from image

### LINKED ISSUE #1017: Cropping Face from Image
# Aim

Cropping faces from image

# Details

- The script added will crop the face from the image using OpenCV and Python

### PR DESCRIPTION
**Pull request by undetectablevirus**
# Description
The script added will crop the faces detected in the image.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x]  I have added the project meta data in the PR template.
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

## Project Metadata
- [x] None

Title: Cropping Faces from Image

Folder: Cropping Faces from Image

Requirements:  Python, OpenCV and Trained classifier XML file in your machine 

Script:  Cropping Faces from image.py

Arguments: none

Contributor: Garima Goyal

Description: A script that will allow us to crop faces from a given image.

### CODE REVIEW

File: Cropping faces from image/Readme.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+## Output
+<a href="https://imgbb.com/"><img src="https://i.ibb.co/g949YCs/ss.png" alt="ss" border="0"></a>
+
+
+## Author
+
+Garima Goyal
+
+
```
Comment (kaustubhgupta) [inline Cropping faces from image/Readme.md:18]: Remove one blank line

File: Cropping faces from image/Readme.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+User needs to have Python and OpenCV in their machines.
+
+## Output
+<a href="https://imgbb.com/"><img src="https://i.ibb.co/g949YCs/ss.png" alt="ss" border="0"></a>
+
+
+## Author
+
+Garima Goyal
```
Comment (kaustubhgupta) [inline Cropping faces from image/Readme.md:16]:
```suggestion
[Garima Goyal](https://github.com/undetectablevirus)
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
- Add requirements.txt file
- Add haarcascade_frontalface_default.xml file
- Save the cropped faces as an image file after imshow

 Readme changes:

**Comment by undetectablevirus** [comment]
Will update with the asked changes soon. Thank you for your cooperation.

**Comment by kaustubhgupta** [comment]
> Will update with the asked changes soon. Thank you for your cooperation.

updates on this?

**Comment by undetectablevirus** [comment]
@kaustubhgupta I have done the asked changes. Kindly review and let me know if anything else is needed.

File: Cropping faces from image/Cropping faces from image.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    faces = img[y:y + h, x:x + w]
+    cv2.imshow('Cropped faces',faces)     #Displaying the cropped faces
+   
+  
+cv2.imshow('Original Image',img)              #Displaying the original Image
+
+#Saving the image of cropped faces
+directory = input('Enter the directory where you want to save the image: ')    #Taking input for desired directory
+os.chdir(directory)
+cv2.imwrite('Cropped faces.jpg',faces)   #saving the image
```
Comment (kaustubhgupta) [inline Cropping faces from image/Cropping faces from image.py:32]: Save the image in this format: Image name (Cropped-Faces).jpg
For example:
input image: abc.jpg
output image: abc (Cropped-Faces).jpg

Also fix this path:

```suggestion
cv2.imwrite('./Cropping faces from image/Cropped faces.jpg',faces)   #saving the image
```

File: Cropping faces from image/Requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Firstly, Make sure you have Python installed in your Machine. 
+Second requirement is the installation of OpenCV in your PC. You can install it using 'pip install opencv-python' if not installed.
+For the Haar cascades to work,user needsto download the trained classifier XML file (haarcascade_frontalface_default.xml), which is available in OpenCv’s GitHub repository. 
+Save this file to your working location.
```
Comment (kaustubhgupta) [inline Cropping faces from image/Requirements.txt:4]: This is not the way of defining the requirements.txt file. Take a look at this file:
https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Amazon-Price-Alert/requirements.txt

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Update the PR template to add the project metadata. Refer to this PR for reference: https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/1002

Also, more changes:

File: Cropping faces from image/Requirements.txt (hunk 4 above)
Comment (undetectablevirus) [reply Cropping faces from image/Requirements.txt:4]: Ok will do these changes asap.

**Comment by undetectablevirus** [comment]
Changes done as asked. Kindly review @kaustubhgupta

File: Cropping faces from image/Cropping faces from image.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+img = cv2.imread(path)
+  
+# Convert image into grayscale
+gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)   
+
+#cascade loading
+face_cascade = cv2.CascadeClassifier('haarcascade_frontalface_alt2.xml')
+
+face_cascade = cv2.CascadeClassifier(cv2.data.haarcascades + 'haarcascade_frontalface_default.xml')
```
Comment (kaustubhgupta) [inline Cropping faces from image/Cropping faces from image.py:15]: Why this is loaded two times? Also add the file `haarcascade_frontalface_default.xml` used in this script

File: Cropping faces from image/Requirements.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+python=3.8.2
+opencv=opencv_version =4.4.0.46
```
Comment (kaustubhgupta) [inline Cropping faces from image/Requirements.txt:2]: wrongly written. single `=` is not correct. Also, can you run this file using the command `pip install requirements.txt`? The file name is wrong. It is requirements.txt and not Requirements.txt.

File: Cropping faces from image/Requirements.txt [hunk 7]
```diff
@@ -0,0 +1,3 @@
...
+Trained Calssifier XML File=haarcascade_frontalface_default.xml
```
Comment (kaustubhgupta) [inline Cropping faces from image/Requirements.txt:3]: This does not count under requirements. See the file again which I referred

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by kaustubhgupta** [comment]
@undetectablevirus updates on this?

**Comment by kaustubhgupta** [comment]
PR closed as participant showed no activity

**Pull request closed without merging**